src/argilla/client/client.py

```diff
@@ -208,6 +215,50 @@ def get_workspace(self) -> str:
         """
         return self._client.headers.get(WORKSPACE_HEADER_NAME)
 
+    def list_workspaces(self) -> List[WorkspaceModel]:
+        """Lists all the availble workspaces for the current user.
+
+        Returns:
+            A list of `WorkspaceModel` objects, containing the workspace
+            attributes: name, id, created_at, and updated_at.
+        """
+        return workspaces_api.list_workspaces(client=self._client).parsed
+
+    def list_datasets(self) -> List[FeedbackDatasetModel]:
+        return datasets_api_v1.list_datasets(client=self._client).parsed
+
+    def create_dataset(self, name: str, workspace_id: str, guidelines: Optional[str] = None) -> httpx.Response:
+        return datasets_api_v1.create_dataset(
+            client=self._client, name=name, workspace_id=workspace_id, guidelines=guidelines
+        ).parsed
+
+    def get_dataset(self, id: str) -> FeedbackDatasetModel:
+        return datasets_api_v1.get_dataset(client=self._client, id=id).parsed
+
+    def delete_dataset(self, id: str) -> httpx.Response:
+        return datasets_api_v1.delete_dataset(client=self._client, id=id)
+
+    def publish_dataset(self, id: str) -> FeedbackDatasetModel:
+        return datasets_api_v1.publish_dataset(client=self._client, id=id).parsed
+
+    def get_records(self, id: str, offset: int = 0, limit: int = 50) -> FeedbackRecordsModel:
+        return datasets_api_v1.get_records(client=self._client, id=id, offset=offset, limit=limit).parsed
+
+    def add_records(self, id: str, records: Union[Dict[str, Any], List[Dict[str, Any]]]) -> None:
+        return datasets_api_v1.add_records(client=self._client, id=id, records=records)
+
+    def get_fields(self, id: str) -> List[Dict[str, Any]]:
+        return datasets_api_v1.get_fields(client=self._client, id=id).parsed
+
+    def add_field(self, id: str, field: Dict[str, Any]) -> httpx.Response:
+        return datasets_api_v1.add_field(client=self._client, id=id, field=field)
+
+    def get_questions(self, id: str) -> List[Dict[str, Any]]:
+        return datasets_api_v1.get_questions(client=self._client, id=id).parsed
+
+    def add_question(self, id: str, question: Dict[str, Any]) -> httpx.Response:
+        return datasets_api_v1.add_question(client=self._client, id=id, question=question)
+
```

Comment:
All these methods are referring to the new apiV1 endpoints and could be confusing to keep them here as is. For example, the `list_datasets` method will return only new kinds of datasets. 

Could we move them to a DatasetsV1 class and work with them from there?

```python 
class DatasetsV1:

   def __init__(client: AuthenticatedCliewnt):
       self.client = client

  def list_datasets(self) -> List[FeedbackDatasetModel]:
        return datasets_api_v1.list_datasets(client=self._client).parsed
```

And then use as:

```python
argilla_client.datasets_v1.list_datasets()
```